### ISSUE TITLE
Allow regex for allowModules

### ISSUE THREAD

**Issue by gpoitch** [mysticatea/eslint-plugin-node#270]
Ability to use a string or regex for allowModules.  For example, say I have a directory I alias in typescript/webpack/whatever to `@foo`, I want to to allow everything in that directory:
```js
settings: {
    node: {
      allowModules: [/^@foo\//', 'bar']
    }
}
```

This should help address some other issues such as #217, #233 in a generic way.
